symengine/cwrapper.cpp

```diff
@@ -399,9 +399,17 @@ void basic_pow(basic s, const basic a, const basic b)
     s->m = SymEngine::pow(a->m, b->m);
 }
 
-void basic_div(basic s, const basic a, const basic b)
+int basic_div(basic s, const basic a, const basic b)
 {
-    s->m = SymEngine::div(a->m, b->m);
+    try
+    {
+        s->m = SymEngine::div(a->m, b->m);
+        return 0;
+    }
+    catch ( ... )
```

Comment:
`NotImplementedError` for not implemented things.
`DomainError` for wrong domains, like evaluating complexes in real domain.
I think you will need to add more when you go through the codebase.